src/protocol/requests.rs

```diff
@@ -98,6 +100,14 @@ pub struct TextDocumentParams {
     pub text_document: TextDocument,
 }
 
+#[derive(Serialize, Deserialize, Clone)]
+pub struct TextDocumentChangeParams {
```

Comment:
I would change it to DidChangeTextDocumentParams, so it completely matches https://github.com/microsoft/language-server-protocol/blob/master/versions/protocol-2-x.md#textDocument_didChange

Reply:
I don't think the long names just to match the typescript implementation add much value as long as the names are descriptive. Also we could just add comments to point to their typescript counter part.